src/json.path.ts

```diff
@@ -87,9 +87,14 @@ export function getPropertyPathWithQuotes(
   return `[${propertyNameJson}]`;
 }
 
-export const nonQuotedCharacterRanges = ['A-Z', 'À-Ö', 'Ø-ö', 'ø-ÿ'];
+export const nonQuotedCharacterRanges = ['A-Z', 'À-Ö', 'Ø-ö', 'ø-ÿ', '0-9'];
 
 export function propertyRequiresQuotes(propertyName: jsonc.Segment): boolean {
+  // If the property start with a numbrer we are forced to require quotes
+  if (propertyName.toString().match(/^\d/)) {
+    return true
```

Comment:
```suggestion
    return true;
```
Just to be consistent with the code base :sweat_smile: